src/Guide/Api/Utils.hs

```diff
@@ -89,3 +94,43 @@ instance (HasSwagger api, KnownSymbol name, KnownSymbol desc) =>
              symbolVal (Proxy @desc))
             Nothing
      in toSwagger (Proxy @api) & applyTags [tag]
+
+
+-- | Type to log user's edition with header details
+data RequestDetails = RequestDetails
+  { rdIp      :: Text
+  , rdReferer :: Text
+  , rdUA      :: Text
+  } deriving (Show, Generic)
+
+instance (HasServer api context)
+  => HasServer (Maybe RequestDetails :> api) context where
+
+  type ServerT (Maybe RequestDetails :> api) m = Maybe RequestDetails -> ServerT api m
```

Comment:
I get the point of `Maybe` here but it's rather unusual for a Servant combinator.

It's better to have just `RequestDetails` and make the fields optional. For some handlers we might need only some fields, for instance, and for some handlers we might need all of them. The handler should decide.